src/ngraph/pass/graph_rewrite.cpp

```diff
@@ -64,35 +63,28 @@ bool pass::GraphRewrite::run_on_function(shared_ptr<Function> f)
     bool rewritten = false;
     const size_t NUM_TRIES = 10;
     size_t tries = NUM_TRIES;
-    vector<shared_ptr<pattern::Matcher>> original_matchers{m_matchers};
-    bool is_dynamic_function = f->is_dynamic();
+    vector<MatchClosure> original_matchers{m_matchers};
     do
     {
         rewritten = false;
-        vector<shared_ptr<pattern::Matcher>> matchers{m_matchers};
+        // m_matchers may contain newly constructed matchers for matchers
+        // that need multiple passes. See comments above.
+        vector<MatchClosure> run_matchers{m_matchers};
         m_matchers.clear();
         for (auto node : f->get_ordered_ops())
         {
-            for (auto matcher : matchers)
+            for (auto& mc : run_matchers)
             {
-                if (is_dynamic_function &&
-                    (matcher->get_property(pass::PassProperty::REQUIRE_STATIC_SHAPE)))
```

Comment:
Do we want to keep the `REQUIRE_STATIC_SHAPE` logic? (Has it been moved elsewhere?)

Reply:
I got rid of them because there's enough change in this PR that I need to make some adjustments to the GraphRewrite logic on handling dynamic shape. That's going to be in another PR.